src/EnergyPlus/PlantComponent.hh

```diff
@@ -0,0 +1,20 @@
+#ifndef PLANTCOMPONENT_HH_INCLUDED
+#define PLANTCOMPONENT_HH_INCLUDED
+
+#include <string>
+#include <memory>
+
+class PlantComponent
+{
+public:
+	std::string name;
+	int compType;
```

Comment:
This is curious to me.  In some places I see types identified by string.  (Like up in the factory for pipe.)  In other places there is string and int.  (Like here https://github.com/NREL/EnergyPlus/pull/4945/files#diff-ae587fad7fd53d7ab16d3d93e273bcefR182 and here https://github.com/NREL/EnergyPlus/pull/4945/files#diff-d8cdb0f9f5f04513135a9405b38ec78dR11.  I think I understand the drivers.  When you're coming from idf input you are talking in terms of strings, but when you get inside you probably want the int for fast comparison.  Polymorphism might reduce the need for determining type, but I imagine it will still be around a fair bit.  How about unifying these into a new type like [IddObjectType](https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/utilities/idd/IddEnums.hpp#L105) that has string integer bound together?